### ISSUE TITLE
Broken aws-code-deploy@0.0.11 (Succeeded deploy fails)

### ISSUE THREAD

**Issue by kml** [CircleCI-Public/circleci-orbs#277]
### Orb version

aws-code-deploy@0.0.11

### What happened

Logs:
```
Status: InProgress...
Status: InProgress...
Deployment failed!
{
    "deploymentInfo": {
        "applicationName": "my_app", 
        "status": "Succeeded", 
```

### Expected behavior

Success

https://docs.aws.amazon.com/codedeploy/latest/APIReference/API_DeploymentOverview.html
"Succeeded" status it treated as failure
Relates to https://github.com/CircleCI-Public/circleci-orbs/pull/273

**Comment by kml** [comment]
@felicianotech

**Comment by iynere** [comment]
@kml thank you! sounds like this results from [this pr](https://github.com/CircleCI-Public/circleci-orbs/pull/236)

@kobim am I misunderstanding something here? "Succeeded" certainly sounds like it should also count as a success

**Comment by kobim** [comment]
@iynere sounds correct.
[seems like](https://docs.aws.amazon.com/codedeploy/latest/APIReference/API_DeploymentInfo.html#CodeDeploy-Type-DeploymentInfo-status) `Ready` is valid only in blue/green deployments (and we can assume this is successful and no need to further wait on the deployment), and `Succeeded` will always be the final and successful status.

**Comment by shawnjohnson** [comment]
@iynere  - It looks like this issue can be closed now that #278 is merged?  I was able to try it out with the new version 0.0.12 and it worked successfully with a `CodeDeployDefault.OneAtATime` type deployment.

CircleCI output:
```
if [[ $STATUS == "Succeeded" ]]; then
  EXITCODE=0
  echo "Deployment finished."
else
  EXITCODE=1
  echo "Deployment failed!"
fi
aws deploy get-deployment --deployment-id $ID
exit $EXITCODE
Status: Created...
Status: Created...
Status: InProgress...
Status: InProgress...
Status: InProgress...
Deployment finished.
{
    "deploymentInfo": {
        "applicationName": "my-app", 
        "status": "Succeeded", 
        "deploymentOverview": {
            "Skipped": 0, 
            "Succeeded": 1, 
            "Failed": 0, 
            "Ready": 0, 
            "InProgress": 0, 
            "Pending": 0
        }, 
```

**Comment by iynere** [comment]
fantastic, thank you!
